ext/date/php_date.c

```diff
@@ -3427,6 +3427,12 @@ static int timezone_initialize(php_timezone_obj *tzobj, const char *tz, size_t t
 	}
 
 	dummy_t->z = timelib_parse_zone(&tz, &dst, dummy_t, &not_found, DATE_TIMEZONEDB, php_date_parse_tzfile_wrapper);
+	if (!not_found && (*tz != '\0')) {
```

Comment:
Checking `tz_len != 0` sounds more intuitive (and is probably faster too).

Reply:
No, `tz_len` is the original full length of the string. `timelib_parse_zone` modifies `&tz` to the end of the string, or when it finds something odd. If it finds something odd, `*tz != '\0'`. If it doesn't, it has reach the end of the string. It's a stupid API, but we'll have to live with it.

It's the situation when this happens:
```
"UTC xx"
```
The parser finds:
```
"UTC"
```
And then the `tz` is still at:
```
" xx"
```

Reply:
Ah ok, I didn't check if `timelib_parse_zone` modifies `tz`.